CONTRIBUTING.md

```diff
@@ -126,3 +126,49 @@ We encourage contributors to leverage Large Language Models (LLMs) responsibly:
   ```sh
   sbt "scalafmtCheckAll"
   ```
+
+## Design Guidelines
+
+### Adding a New Method
+
+If you want to contribute a new method like `S.newMethod` or `s.newMethod`, feel free to:
+
+* Open an issue
+    
+* Discuss on Discord: [https://discord.gg/afch62vKaW](https://discord.gg/afch62vKaW)
+    
+* Share design examples:
+    
+    * Use cases
+        
+    * Equivalent in `ZIO` or `Cats Effect`
+        
+    * Other motivating patterns
+        
+
+### Where to add Your `newMethod`
+
+| Subproject | Use For |
+| --- | --- |
+| `kyo-data` | Data structures (`Chunk`, `Maybe`, `Result`, etc.) |
+| `kyo-prelude` | Effect types without `IO` (`Abort`, `Env`, `Var`, etc.) |
+| `kyo-core` | Methods requiring `IO`, `Async`, or `Resource` |
+| `kyo-combinators` | Extensions or composition helpers |
+
+Add corresponding tests in the same subproject.
+
+**Example:**  
+A new `Stream.fromSomething` method:
+
+* If it uses `IO`: place it in `kyo-core/shared/src/main/scala/kyo/StreamCoreExtensions.scala`
+    
+* If it doesn't: place it in `kyo-prelude/shared/src/main/scala/kyo/Stream.scala`
+    
+
+### Tips
+
+* Prefer [`call-by-name`](https://docs.scala-lang.org/tour/by-name-parameters.html) (`body: => A < S`) for deferred evaluation, especially when lifting to `IO`.
+    
+* Use [`inline`](https://docs.scala-lang.org/scala3/reference/metaprogramming/inline.html) when beneficial.
+    
+* Keep `S` open: use `A < (S & SomeEffect)` instead of `A < SomeEffect` to support composition and emulate effect row polymorphism.
```

Comment:
I don't think `emulate effect row polymorphism` is accurate. Kyo's tracking of effects isn't an emulation of row polymorphism, it's a distinct and proper tracking mechanism based on Scala's type system. It provides more advanced behaviors like parametrized effects with variance, which mixed with other type system features like union types I'd argue provides more flexibility and better usability than row polymorphism.